### PR TITLE
docs: Update AlloyDB setup guide to reflect Public IP support + doc updates

### PR DESCRIPTION
**Pull request by anubhav756**

### CODE REVIEW

File: docs/datastore/alloydb.md [hunk 1]
```diff
@@ -47,9 +47,10 @@
 [install-alloydb-auth-proxy]: https://cloud.google.com/alloydb/docs/auth-proxy/connect#install
 
 
-## Enable private services access
+## Enable private services access (Non-Cloud Postgres)
```
Comment (kurtisvg) [inline docs/datastore/alloydb.md:50]: Why do we need to enable PSA for non-cloud postgres?

**Review by kurtisvg — APPROVED** [review]

File: docs/datastore/alloydb.md (hunk 1 above)
Comment (kurtisvg) [inline docs/datastore/alloydb.md:50]: Why do we need to enable PSA for non-cloud postgres?

**Review by kurtisvg — CHANGES_REQUESTED** [review]

File: docs/datastore/alloydb.md (hunk 1 above)
Comment (anubhav756) [reply docs/datastore/alloydb.md]: Updated the commit 5f07874 with the changes suggested.

File: docs/datastore/alloydb.md [hunk 2]
```diff
@@ -254,3 +260,14 @@ gcloud builds submit --config retrieval_service/alloydb.tests.cloudbuild.yaml \
 ```
 
 Where `$DB_NAME`,`$DB_USER`,`$DB_REGION`,`$DB_CLUSTER`,`$DB_INSTANCE` are environment variables with your database values.
+
+### Troubleshoot
```
Comment (kurtisvg) [inline docs/datastore/alloydb.md:264]:
```suggestion
### Troubleshooting

```

File: docs/datastore/postgres.md [hunk 3]
```diff
@@ -43,73 +43,11 @@
 [venv]: https://cloud.google.com/python/docs/setup#installing_and_using_virtualenv
 [install-psql]: https://www.timescale.com/blog/how-to-install-psql-on-mac-ubuntu-debian-windows/
 
+## Setup Postgres database
 
-## Enable private services access
+This section assumes you've already set up a non-cloud database like [AlloyDB Omni](https://cloud.google.com/alloydb/docs/omni).
```
Comment (kurtisvg) [inline docs/datastore/postgres.md:48]: Can we make this link to the AlloyDB Omni install / quickstart instructions?

**Review by kurtisvg — APPROVED** [review]
Approved w/ 2 nits

**Pull request merged**